openedx/features/offline_mode/utils.py

```diff
@@ -19,6 +25,32 @@ def get_offline_service_user():
         return User.objects.get(username=settings.OFFLINE_SERVICE_WORKER_USERNAME)
     except User.DoesNotExist as e:
         log.error(
-            f'Service user with username {settings.OFFLINE_SERVICE_WORKER_USERNAME} to render XBlock does not exist.'
+            f"Service user with username {settings.OFFLINE_SERVICE_WORKER_USERNAME} to render XBlock does not exist."
         )
         raise e
+
+
+def clear_deleted_content(course_key):
+    """
+    Delete the offline content archive for the blocks that are deleted from the course.
+    """
+    all_course_offline_archive_names = {
```

Comment:
I think it will be a bit more optimized
```suggestion
    base_offline_course_path = settings.OFFLINE_CONTENT_PATH_TEMPLATE.format(course_id=str(course_key))
    if default_storage.exists(base_offline_course_path):
        _, file_names = default_storage.listdir(base_offline_course_path)
    else:
        return

    all_course_offline_archive_names = {
        f"{xblock.location.block_id}.zip"
        for block_type in OFFLINE_SUPPORTED_XBLOCKS
        for xblock in modulestore().get_items(course_key, qualifiers={"category": block_type})
    }     
```